File: macos/Scanner/Sources/Scanner/Rules/AESMCP008.swift
public struct AESMCP008: Rule {
    public let id = "AES-MCP-008"
    public let severity: Severity = .high

    public init() {}

    public func evaluate(_ facts: ScanFacts) -> [Finding] {
        facts.mcpServers.compactMap { server in
            guard !server.disabled else {
                return nil
            }
            let overridden = server.env.contains { key, value in
                AiThreatPatterns.isModelBaseUrlOverride(key: key, value: value)
            }
            guard overridden else {
                return nil
            }
            return finding(app: server.appId, serverName: server.name, affectedPath: server.configPath)
        }
    }
}

File: windows/Scanner/Rules/AESMCP008.cs
using AIExposureScanner.Scanner.Models;

namespace AIExposureScanner.Scanner.Rules;

public sealed class AESMCP008 : IRule
{
    public string Id => "AES-MCP-008";
    public Severity Severity => Severity.High;

    public IReadOnlyList<Finding> Evaluate(ScanFacts facts) =>
        facts.McpServers
            .Where(server => !server.Disabled)
            .Where(server => server.Env.Any(kv => AiThreatPatterns.IsModelBaseUrlOverride(kv.Key, kv.Value)))
            .Select(server => this.Finding(server.AppId, server.Name, affectedPath: server.ConfigPath))
            .ToArray();
}
